smithy-model/src/test/java/software/amazon/smithy/model/transform/RenameShapesTest.java

```diff
@@ -0,0 +1,268 @@
+import software.amazon.smithy.model.shapes.Shape;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.shapes.StringShape;
+import software.amazon.smithy.model.shapes.StructureShape;
+import software.amazon.smithy.model.shapes.UnionShape;
+import software.amazon.smithy.model.traits.Trait;
+
+public class RenameShapesTest {
+
+    @Test
+    public void returnsUnmodifiedModelIfGivenEmptyRenameMapping() {
+        ShapeId stringId = ShapeId.from("ns.foo#String");
+        StringShape fooTarget = StringShape.builder().id(stringId).build();
+        Model model = Model.builder()
+                .addShapes(fooTarget)
+                .build();
+        ModelTransformer transformer = ModelTransformer.create();
+        Map<ShapeId, ShapeId> renamed = new HashMap<>();
+        Model result = transformer.renameShapes(model, renamed);
+
+        assertThat(result.shapes().count(), Matchers.equalTo(1L));
+        assertThat(result.getShape(stringId).get(), Matchers.is(fooTarget));
+    }
+
+    @Test
+    public void returnsUnmodifiedModelIfToAndFromAreEqual() {
+        ShapeId stringId = ShapeId.from("ns.foo#String");
+        StringShape target = StringShape.builder().id(stringId).build();
+        Model model = Model.builder()
+                .addShapes(target)
+                .build();
+        ModelTransformer transformer = ModelTransformer.create();
+        Map<ShapeId, ShapeId> renamed = new HashMap<>();
+        renamed.put(stringId, stringId );
+        Model result = transformer.renameShapes(model, renamed);
+
+        assertThat(result.shapes().count(), Matchers.equalTo(1L));
+        assertThat(result.getShape(stringId).get(), Matchers.is(target));
+    }
+
+    @Test
+    public void returnsModelWithRenamedStringShape() {
+        ShapeId fromStringId = ShapeId.from("ns.foo#String");
+        ShapeId containerId = ShapeId.from("ns.foo#Container");
+        ShapeId keyId = ShapeId.from("ns.foo#Container$key");
+        ShapeId valueId = ShapeId.from("ns.foo#Container$value");
+
+        StringShape target = StringShape.builder().id(fromStringId).build();
+        MemberShape keyMember = MemberShape.builder().id(keyId).target(fromStringId).build();
+        MemberShape valueMember = MemberShape.builder().id(valueId).target(fromStringId).build();
+        MapShape container = MapShape.builder().id(containerId).key(keyMember).value(valueMember).build();
+        Model model = Model.builder()
+                .addShapes(target, keyMember, valueMember, container)
+                .build();
+        ModelTransformer transformer = ModelTransformer.create();
+
+        ShapeId toStringId = ShapeId.from("ns.bar#String");
+        Map<ShapeId, ShapeId> renamed = new HashMap<>();
+        renamed.put(fromStringId, toStringId );
+        Model result = transformer.renameShapes(model, renamed);
+
+        assertThat(result.getShape(toStringId).isPresent(), Matchers.is(true));
```

Comment:
nit: usually it's clearer (and consistent with the rest of our codebase) to just use a static import for Matchers.